### PR TITLE
Add form to edit personal info

### LINKED ISSUE #427: Add a city field to User model
1. add `city : string | null` field to the User model
2. add an input box on /profile page to update city they live in. label text should be "居住的中国城市或者国家+城市”
3. add another input box to update `wexin`
4. another to update `sex`. give only two options `男` and `女`

Note that the /profile page was developed very long time ago and may need some code refactoring before adding new functions. For example, it should be updated to use EditableWithIconOrLink for text fields. The PR that refactors existing code should be separate form the PR that adds new logic.

### PR DESCRIPTION
**Pull request by ada-only-one-ada**
Fixed #427 
Screenshot 2024-10-24 at 16 39 20

### CODE REVIEW

**Comment by weihanwang** [comment]
1. 不需要有第一个截图里的编辑按钮。这页打开以后直接展示第二个截图的内容就成。
2. 页面太宽。所有控件应该被包裹在一个container里，限制在一个比较合适的宽度。

**Comment by ada-only-one-ada** [comment]
Cannot figure out why preview fails.

**Comment by weihanwang** [comment]
1. the form should be flushed to the left 
2. use a readonly input box for email address, so it appears similarly to the other fields
3. use white background for all other input boxes
4. move wechat to before name and after email
5. remove the Cancel button

File: src/pages/profile.tsx [hunk 1]
```diff
@@ -1,123 +1,90 @@
...
-      <EmailField />
-      <NameField />
+      <FormControl>
+        <FormLabel>微信</FormLabel>
+        <Input value={newWechat} onChange={e => setNewWechat(e.target.value)} />
+      </FormControl>               
+    
+      <Flex gap={componentSpacing}>
+        <Button onClick={handleCancel}>取消</Button>
+        <Button onClick={handleSubmit} colorScheme="green">保存</Button>
```
Comment (weihanwang) [inline src/pages/profile.tsx:85]: use `variant="brand"`

**Review by weihanwang — APPROVED** [review]

**Pull request merged**